internal/snmp/wrapper.go

```diff
@@ -167,6 +167,14 @@ func (gs *GosnmpWrapper) SetAgent(agent string) error {
 	switch u.Scheme {
 	case "tcp":
 		gs.Transport = "tcp"
+	case "tcp4":
+		gs.Transport = "tcp4"
+	case "tcp6":
+		gs.Transport = "tcp6"
+	case "udp4":
+		gs.Transport = "udp4"
+	case "udp6":
+		gs.Transport = "udp6"
 	case "", "udp":
```

Comment:
I assume the empty string can be removed?
```suggestion
	case "udp":
```

Reply:
Gosnmp will default to "udp" if no scheme is specified. 
However, in userconfig we can omit the "udp://" which will then cause the switch-case to fall through to the default-case. 

Regarding #7879:
The default case is reached when an invalid scheme is specified, but telegraf does not output the error nor stop execution. I will look into it further.

Reply:
Correct, but isn't that handled by this part? https://github.com/influxdata/telegraf/blob/3b8df55b9c8c15353b8778ab3a19f0b1a5021d29/internal/snmp/wrapper.go#L158-L160

Reply:
yes, you right. 
I removed the line.